lib/rspec/mocks/exclude_stubbed_classes_from_subclasses.rb

```diff
@@ -0,0 +1,56 @@
+module RSpec
+  module Mocks
+    # Support for `exclude_stubbed_classes_from_subclasses` configuration.
+    #
+    # @private
+    class ExcludeStubbedClassesFromSubclasses
+      class << self
+        def enable!
+          return unless RUBY_VERSION >= "3.1"
+          return if Class.respond_to?(:subclasses_with_rspec_mocks)
+
+          Class.class_eval do
+            def subclasses_with_rspec_mocks
+              subclasses_without_rspec_mocks - RSpec::Mocks::ExcludeStubbedClassesFromSubclasses.excluded_subclasses
+            end
+
+            alias subclasses_without_rspec_mocks subclasses
+            alias subclasses subclasses_with_rspec_mocks
+          end
+        end
+
+        def disable!
+          @excluded_subclasses = []
+
+          if Class.respond_to?(:subclasses_with_rspec_mocks)
+            Class.class_eval do
+              undef subclasses_with_rspec_mocks
+              alias subclasses subclasses_without_rspec_mocks
+              undef subclasses_without_rspec_mocks
+            end
+          end
+        end
+
+        def excluded_subclasses
+          require 'weakref' unless defined?(::WeakRef)
```

Comment:
Ok to move this line into `enable!`?

Reply:
I'm not sure if it's a problem but now we can do `RSpec::Mocks::ExcludeStubbedClassesFromSubclasses.excluded_subclasses` without `ExcludeStubbedClassesFromSubclasses.enable!` first. If we move this line to `enable!`, it will raise an error.

Reply:
As you wish, I can do : 


```
diff --git a/lib/rspec/mocks/exclude_stubbed_classes_from_subclasses.rb b/lib/rspec/mocks/exclude_stubbed_classes_from_subclasses.rb
index bb72101e..129ff0d2 100644
--- a/lib/rspec/mocks/exclude_stubbed_classes_from_subclasses.rb
+++ b/lib/rspec/mocks/exclude_stubbed_classes_from_subclasses.rb
@@ -8,6 +8,8 @@ module RSpec
         def enable!
           return unless RUBY_VERSION >= "3.1"
           return if Class.respond_to?(:subclasses_with_rspec_mocks)
+          @excluded_subclasses = []
+          require 'weakref' unless defined?(::WeakRef)
 
           Class.class_eval do
             def subclasses_with_rspec_mocks
@@ -32,9 +34,6 @@ module RSpec
         end
 
         def excluded_subclasses
-          require 'weakref' unless defined?(::WeakRef)
-
-          @excluded_subclasses ||= []
           @excluded_subclasses.select(&:weakref_alive?).map do |ref|
             begin
               ref.__getobj__
@@ -45,9 +44,6 @@ module RSpec
         end
 
         def exclude_subclass(constant)
-          require 'weakref' unless defined?(::WeakRef)
-
-          @excluded_subclasses ||= []
           @excluded_subclasses << ::WeakRef.new(constant)
         end
       end
diff --git a/spec/rspec/mocks/exclude_stubbed_classes_from_subclasses_spec.rb b/spec/rspec/mocks/exclude_stubbed_classes_from_subclasses_spec.rb
index 3b5ded94..16454922 100644
--- a/spec/rspec/mocks/exclude_stubbed_classes_from_subclasses_spec.rb
+++ b/spec/rspec/mocks/exclude_stubbed_classes_from_subclasses_spec.rb
@@ -63,6 +63,10 @@ if RUBY_VERSION >= '3.1'
         end
 
         describe '.excluded_subclasses' do
+          before do
+            described_class.enable!
+          end
+
           it 'returns excluded subclasses' do
             subclass = Class.new
             described_class.exclude_subclass(subclass)
@@ -75,14 +79,12 @@ if RUBY_VERSION >= '3.1'
             described_class.exclude_subclass(subclass)
 
             subclass = nil
-
             GC.start
 
             expect(described_class.excluded_subclasses).to eq([])
           end
 
           it 'does not return excluded subclasses that raises a ::WeakRef::RefError' do
-            require 'weakref'
             subclass = double(:weakref_alive? => true)
             described_class.instance_variable_set(:@excluded_subclasses, [subclass])
 
```